### ISSUE TITLE
Remove nunit from lib and use the NuGet package instead (dev)

### ISSUE THREAD

**Issue by BenedekFarkas** [OrchardCMS/Orchard#6665]
`Orchard.Azure.Tests` has just been consolidated into `Orchard.sln` on the `dev` branch, since `Orchard.Azure.sln` has been deleted. It still uses the `lib` folder for `NUnit` assemblies, while all the other test projects are using the NuGet package. Modifying `Orchard.Azure.Tests` to use NuGet package would make it possible to get rid of `lib\nuget`, but that requires other changes as well.

I'm suggesting the following action plan:
- Add `NUnit` version "2.5.10.11092" to the `packages.config` of `Orchard.Azure.Tests`. Reference to `nunit.framework` in `Orchard.Azure.Tests.csproj` needs to be changed accordingly.
- Change the required version of `Moq` in the the `packages.config` of `Orchard.Azure.Tests` to match the version of the other test projects (4.2.1510.2205 -> 4.0.10827, but that's a downgrade) or do the same thing the other way around (upgrade all the other ones).
- Remove the whole `lib\nunit` folder.
- Adapt `Orchard.proj` to these changes:
    - The `$(LibFolder)\nunit` paths should be changed to use a new property called `NUnitFolder`, which should point to the NuGet package. That's tricky, because the folder name contains the version number and wildcard doesn't work here - maybe looping through the subfolders of `src\packages` to find it?
    - In some cases (e.g. the executable) the path needs to be `$(NUnitFolder)\tools`, because the NuGet package has a different folder structure from `lib\nunit`.
- Fix `Orchard.Azure.Tests` (I couldn't run them, I think one of the problems is that the Storage Emulator's path is now different compared to earlier versions).
    - `AzureVirtualEnvironmentTest.cs` needs some serious code styling. :)

cc: @DanielStolt @sebastienros 

PS: I hope GitHub doesn't charge me for the extra backticks. :D

**Comment by DanielStolt** [comment]
@BenedekFarkas Sounds like a solid plan to me. The reason I added a newer version of `Moq` was thas I did this **before** moving it into the main solution, so I didn't have visibility on what version was already used there. Downgrading to whatever is used there sounds reasonable.

Not sure how to solve the `Orchard.proj` problem though, other than to keep the `lib\nunint` folder around for this purpose.

**Comment by BenedekFarkas** [comment]
Yep, I assumed that was the case. If it's working for `Orchard.Tests.Azure`, then it's probably OK to upgrade all the other ones as well after testing it (just to be sure I downgraded that one for now).

I pushed some changes to the `tests/nunit-azure` branch (https://github.com/OrchardCMS/Orchard/commits/tests/nunit-azure). All tests in `Orchard.Azure.Tests` are passing now. One important thing to note is that I adapted the way the Storage Emulator is started to be able run the tests with the latest version (4.2), but we can add backwards-compatibility, if necessary (maybe for TeamCity build agents for example).

NUnit is still in lib, I'll see if we can remove that.

**Comment by BenedekFarkas** [comment]
Btw full a build (the build target TeamCity executes) is all-green for me (with NUnit still in lib, just for the sake of the test runner executable).

**Comment by DanielStolt** [comment]
Good work @BenedekFarkas - I would pull it in!

**Comment by BenedekFarkas** [comment]
Thanks! :)
I've updated the branch with the following changes:
- `Compile` build target now has a dependency `Test` (doesn't work without it ofc).
- `Orchard.proj` modified to use the NuGet package of `NUnit`. The only trick here is that the build breaks if there's more than one versions of it restored. I think it's a rare case, but not impossible, so I'll see we can fix that.
- Removed the whole `lib\nunit` folder.

Btw when supporting custom solutions for `Orchard.proj` (#6627) is merged into `dev`, we can further adapt the code (with a 1-line addition) to use the packages folder of the solution (which is not necessarily in the `src` folder).

**Comment by DanielStolt** [comment]
Hmm... a rare case that we would actually **target** different versions of NUnit (since all tests are now in the same solution and probably can't be run by the same runner if they target different versions of NUnit, so we would have to always keep NUnit package references completely consolidated) but I think a more frequent case when this would break is when we update the NUnit reference - then you would be sitting with both versions in your `packages` folder, and it wouldn't be very obvious that you have to delete one of them to get the file to validate, so yeah, that would be good to fix.

**Comment by BenedekFarkas** [comment]
Someone already had exactly the same problem. http://stackoverflow.com/questions/19999786/how-to-get-last-item-from-msbuild-items
